src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2687,104 +2730,212 @@ def _ensure_container_insights_for_monitoring(cmd, addon):
+            resource = resources.get_by_id(
+                workspace_resource_id, '2015-11-01-preview')
+            location = resource.location
+        except HttpResponseError as ex:
+            raise ex
+
+    if aad_route:
+        cluster_resource_id = f"/subscriptions/{cluster_subscription}/resourceGroups/{cluster_resource_group_name}/providers/Microsoft.ContainerService/managedClusters/{cluster_name}"
+        dataCollectionRuleName = f"DCR-{workspace_name}"
+        dcr_resource_id = f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/Microsoft.Insights/dataCollectionRules/{dataCollectionRuleName}"
+        from azure.cli.core.util import send_raw_request
+        from azure.cli.core.profiles import ResourceType
+
+        if create_dcr:
+            # first get the association between region display names and region IDs (because for some reason
+            # the "which RPs are available in which regions" check returns region display names)
+            region_names_to_id = {}
+            # retry the request up to two times
+            for _ in range(3):
+                try:
+                    location_list_url = f"https://management.azure.com/subscriptions/{subscription_id}/locations?api-version=2019-11-01"
+                    r = send_raw_request(cmd.cli_ctx, "GET", location_list_url)
+                    break
+                except CLIError as e:
+                    pass
+            else:
+                # This will run if the above for loop was not broken out of. This means all three requests failed
+                raise e
+            json_response = json.loads(r.text)
+            for region_data in json_response["value"]:
+                region_names_to_id[region_data["displayName"]] = region_data["name"]
+
+            # check if region supports DCRs and DCR-A
+            for _ in range(3):
+                try:
+                    feature_check_url = f"https://management.azure.com/subscriptions/{subscription_id}/providers/Microsoft.Insights?api-version=2020-10-01"
+                    r = send_raw_request(cmd.cli_ctx, "GET", feature_check_url)
+                    break
+                except CLIError as e:
+                    pass
+            else:
+                raise e
+            json_response = json.loads(r.text)
+            for resource in json_response["resourceTypes"]:
+                region_ids = map(lambda x: region_names_to_id[x], resource["locations"])  # map is lazy, so doing this for every region isn't slow
+                if resource["resourceType"] == "dataCollectionRules" and location not in region_ids:
+                    raise CLIError(f'Data Collection Rules are not supported for LA workspace region {location}')
+                elif resource["resourceType"] == "dataCollectionRuleAssociations" and cluster_region not in region_ids:
+                    raise CLIError(f'Data Collection Rule Associations are not supported for cluster region {location}')
```

Comment:
Please use a specific error type from `azure.cli.core.azclierror` instead of `CLIError`